test/performance/odp_ml_perf.c

```diff
@@ -0,0 +1,1051 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ * Copyright (c) 2024 Nokia
+ */
+
+/**
+ * @example odp_ml_perf.c
+ *
+ * Performance test application for ML API
+ *
+ * @cond _ODP_HIDE_FROM_DOXYGEN_
+ */
+
+#include <odp_api.h>
+#include <odp/helper/odph_api.h>
+#include <errno.h>
+#include <inttypes.h>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+/* Max number of inputs and outputs */
+#define MAX_IO 8
+
+enum {
+	MODE_INFERENCE = 0,
+	MODE_INFERENCE_QUANT,
+	MODE_CREATE,
+	MODE_LOAD,
+	MODE_NUM,
+};
+
+typedef struct {
+	int mode, num_threads, rounds, latency, warmup;
```

Comment:
It doesn't do any harm, either. The negative range is not used now, but it could be used at some point for some special values, etc. I didn't change these now.

Reply:
Currently, user can pass e.g. negative round and warmup values and tester happily continues, giving weird results. There should at least be some sanity checks after parsing the command line.

Reply:
As a typical user of this tool, I don't need such checks. As a developer, I think if we want to do user input sanitation (and I don't), then we should probably implement it in a library and do it consistently in all of our testing tools, etc.